---
title: "Assignment 1"
author: "Group 6"
date: "9/10/2020"
output: html_document
---


In [4]:
library(e1071)
rice <- read.delim("RiceDataset.txt", header = TRUE, stringsAsFactors = TRUE)

index <- 1:nrow(rice)
set.seed(123)
testindex <- sample(index, trunc(length(index)/3)) 
testset <- rice[testindex,]
trainset <- rice[-testindex,]
svm.model <- svm(CLASS ~ ., data=trainset, type="C-classification", kernel="linear")

results_train <- predict(svm.model,trainset[,-8])
results_test <- predict(svm.model, testset[,-8] )
cf.train <- as.matrix(table(pred=results_train,actual=trainset$CLASS)) 
cf.test <- as.matrix(table(pred=results_test,actual=testset$CLASS))
trainAccuracy = (cf.train)/sum(cf.train)
cf.test

accuracy_train = (cf.train[1,1] + cf.train[2,2])/sum(cf.train)
accuracy_test = (cf.test[1,1] + cf.test[2,2])/sum(cf.test)

          actual
pred       Cammeo Osmancik
  Cammeo      488       42
  Osmancik     51      689

## 1. Interpretation of R code




## 2.Accuracy of R code

In [5]:
accuracy_train = (cf.train[1,1] + cf.train[2,2])/sum(cf.train)
accuracy_test = (cf.test[1,1] + cf.test[2,2])/sum(cf.test)
accuracy_train
accuracy_test

[1] 0.930315

[1] 0.9267717

**Accuracy over on training set:** 93.03%
**Accuracy over test set:** 92.68%

## 3.Using other kernels

### i. Polynomial kernel

In [6]:
svm.model <- svm(CLASS ~ ., data=trainset, type="C-classification", kernel="polynomial")

results_train <- predict(svm.model,trainset[,-8])
results_test <- predict(svm.model, testset[,-8] )
cf.train <- as.matrix(table(pred=results_train,actual=trainset$CLASS)) 
cf.test <- as.matrix(table(pred=results_test,actual=testset$CLASS))
trainAccuracy = (cf.train)/sum(cf.train)
cf.test

accuracy_train = (cf.train[1,1] + cf.train[2,2])/sum(cf.train)
accuracy_test = (cf.test[1,1] + cf.test[2,2])/sum(cf.test)
accuracy_train
accuracy_test

          actual
pred       Cammeo Osmancik
  Cammeo      435       21
  Osmancik    104      710

[1] 0.9106299

[1] 0.9015748

### ii. Sigmoid kernel

In [ ]:
svm.model <- svm(CLASS ~ ., data=trainset, type="C-classification", kernel="sigmoid")

results_train <- predict(svm.model,trainset[,-8])
results_test <- predict(svm.model, testset[,-8] )
cf.train <- as.matrix(table(pred=results_train,actual=trainset$CLASS)) 
cf.test <- as.matrix(table(pred=results_test,actual=testset$CLASS))
trainAccuracy = (cf.train)/sum(cf.train)
cf.test

accuracy_train = (cf.train[1,1] + cf.train[2,2])/sum(cf.train)
accuracy_test = (cf.test[1,1] + cf.test[2,2])/sum(cf.test)
accuracy_train
accuracy_test

### iii. Radial basis kernel

In [ ]:
svm.model <- svm(CLASS ~ ., data=trainset, type="C-classification", kernel="radial")

results_train <- predict(svm.model,trainset[,-8])
results_test <- predict(svm.model, testset[,-8] )
cf.train <- as.matrix(table(pred=results_train,actual=trainset$CLASS)) 
cf.test <- as.matrix(table(pred=results_test,actual=testset$CLASS))
trainAccuracy = (cf.train)/sum(cf.train)
cf.test

accuracy_train = (cf.train[1,1] + cf.train[2,2])/sum(cf.train)
accuracy_test = (cf.test[1,1] + cf.test[2,2])/sum(cf.test)
accuracy_train
accuracy_test

### Summary
#### Accuracy on test set of different kernels:
- Linear: 92.68%
- Polynomial: 90.16%
- Sigmoid: 86.69%
- Radial basis: 92.36%

#### Explanations: